drivers/sensor/bme680/bme680.c

```diff
@@ -0,0 +1,427 @@
+	var2 = (var1 * (s32_t)data->par_t2) >> 11;
+	var3 = ((var1 >> 1) * (var1 >> 1)) >> 12;
+	var3 = ((var3) * ((s32_t)data->par_t3 << 4)) >> 14;
+	data->t_fine = var2 + var3;
+	data->calc_temp = ((data->t_fine * 5) + 128) >> 8;
+}
+
+static void bme680_calc_press(struct bme680_data *data, u32_t adc_press)
+{
+	s32_t var1, var2, var3, calc_press;
+
+	var1 = (((s32_t)data->t_fine) >> 1) - 64000;
+	var2 = ((((var1 >> 2) * (var1 >> 2)) >> 11) *
+		(s32_t)data->par_p6) >> 2;
+	var2 = var2 + ((var1 * (s32_t)data->par_p5) << 1);
+	var2 = (var2 >> 2) + ((s32_t)data->par_p4 << 16);
+	var1 = (((((var1 >> 2) * (var1 >> 2)) >> 13) *
+		 ((s32_t)data->par_p3 << 5)) >> 3)
+	       + (((s32_t)data->par_p2 * var1) >> 1);
+	var1 = var1 >> 18;
+	var1 = ((32768 + var1) * (s32_t)data->par_p1) >> 15;
+	calc_press = 1048576 - adc_press;
+	calc_press = (calc_press - (var2 >> 12)) * ((u32_t)3125);
+	/* This max value is used to provide precedence to multiplication or
+	 * division in the pressure calculation equation to achieve least
+	 * loss of precision and avoiding overflows.
+	 * i.e Comparing value, signed int 32bit (1 << 30)
+	 */
+	if (calc_press >= (s32_t)0x40000000) {
+		calc_press = ((calc_press / var1) << 1);
+	} else {
+		calc_press = ((calc_press << 1) / var1);
+	}
+	var1 = ((s32_t)data->par_p9 *
+		(s32_t)(((calc_press >> 3)
+			 * (calc_press >> 3)) >> 13)) >> 12;
+	var2 = ((s32_t)(calc_press >> 2) * (s32_t)data->par_p8) >> 13;
+	var3 = ((s32_t)(calc_press >> 8) * (s32_t)(calc_press >> 8)
+		* (s32_t)(calc_press >> 8)
+		* (s32_t)data->par_p10) >> 17;
+
+	data->calc_press = calc_press
+			   + ((var1 + var2 + var3
+			       + ((s32_t)data->par_p7 << 7)) >> 4);
+}
+
+static void bme680_calc_humidity(struct bme680_data *data, u16_t adc_humidity)
+{
+	s32_t var1, var2_1, var2_2, var2, var3, var4, var5, var6;
+	s32_t temp_scaled, calc_hum;
+
+	temp_scaled = (((s32_t)data->t_fine * 5) + 128) >> 8;
+	var1 = (s32_t)(adc_humidity - ((s32_t)((s32_t)data->par_h1 * 16))) -
+	       (((temp_scaled * (s32_t)data->par_h3)
+		 / ((s32_t)100)) >> 1);
+	var2_1 = (s32_t)data->par_h2;
+	var2_2 = ((temp_scaled * (s32_t)data->par_h4) / (s32_t)100)
+		 + (((temp_scaled * ((temp_scaled * (s32_t)data->par_h5)
+				     / ((s32_t)100))) >> 6) / ((s32_t)100))
+		 +  (s32_t)(1 << 14);
+	var2 = (var2_1 * var2_2) >> 10;
+	var3 = var1 * var2;
+	var4 = (s32_t)data->par_h6 << 7;
+	var4 = ((var4) + ((temp_scaled * (s32_t)data->par_h7) /
+			  ((s32_t)100))) >> 4;
+	var5 = ((var3 >> 14) * (var3 >> 14)) >> 10;
+	var6 = (var4 * var5) >> 1;
+	calc_hum = (((var3 + var6) >> 10) * ((s32_t)1000)) >> 12;
+
+	if (calc_hum > 100000) { /* Cap at 100%rH */
+		calc_hum = 100000;
+	} else if (calc_hum < 0) {
+		calc_hum = 0;
+	}
+
+	data->calc_humidity = calc_hum;
+}
+
+static void bme680_calc_gas_resistance(struct bme680_data *data, u8_t gas_range,
+				       u16_t adc_gas_res)
+{
+	s64_t var1, var3;
+	u64_t var2;
+
+	u32_t look_up1[16] = { 2147483647, 2147483647, 2147483647, 2147483647,
```

Comment:
Can you make these lookups static const? Where do the magic numbers come from?

Reply:
This can be changed to static const. As mentioned in a different comment, the origin of the magic numbers is confidential.

Reply:
The magic numbers are mentioned in the datasheet.

Reply:
Please add a comment that notes which section in the datasheet the magic numbers are documented.